### PR TITLE
[Coop] Monitor.

### PR DESCRIPTION
**Pull request by jaykrell**

### CODE REVIEW

File: mono/metadata/monitor.c [hunk 1]
```diff
@@ -1172,43 +1178,64 @@ mono_monitor_threads_sync_members_offset (int *status_offset, int *nest_offset)
 	*nest_offset = ENCODE_OFF_SIZE (MONO_STRUCT_OFFSET (MonoThreadsSync, nest), sizeof (ts.nest));
 }
 
-void
-ves_icall_System_Threading_Monitor_Monitor_try_enter_with_atomic_var (MonoObject *obj, guint32 ms, MonoBoolean *lockTaken)
+static void
+ves_icall_System_Threading_Monitor_Monitor_try_enter_with_atomic_var (MonoObject *obj, guint32 ms, MonoBoolean *lockTaken, MonoError* error)
 {
+	// The use of error here is unusual, but expedient, and easy enough to understand.
```
Comment (vargaz) [inline mono/metadata/monitor.c]: Shouldn't this be a MonoObjectHandle obj ?

Comment (vargaz) [reply mono/metadata/monitor.c]: nm.

Comment (vargaz) [inline mono/metadata/monitor.c]: This should be renamed as its no longer an icall function.

**Comment by vargaz** [comment]
This might have a perf impact.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Pull request merged**